### DISCUSSION TITLE
get value of one parameter and pass to another in autocomplete

### DISCUSSION THREAD

**Discussion by aurkaxi** [Rapptz/discord.py#9190] [Q&A] [answered]
There are 2 boxes, they contain:
box(x) { a, b
box(y) { c, d

there is a function(g): `async def function_g(self, interaction):` which returns the name of box x and y
another funtion(h): `async def function_h(self, interaction, in_box): which returns the items of the selected box

so if the decorator goes like `@app_commands.autocomplete(box=function_g, items= function_h(in_box=selectedbox)`

now what do i write for `in_box` parameter in the decorator. (the `selectedbox`). I need the value of `box` before the interaction invokes.

**Comment by ElBe-Plaq** [comment]
Does it work if you use`box=function_g()` instead of `box=function_g`?

**Reply by aurkaxi** [reply]
no, but i solved this with "namespace"

**Reply by ElBe-Plaq** [reply]
So it's answered?

**Reply by aurkaxi** [reply]
Yes sir

**Reply by ElBe-Plaq** [reply]
Can you share your improved code as an answer and mark it as answered?

**Reply by aurkaxi** [reply]
Done

**Comment by aurkaxi** [comment, marked as answer]
```py
class CogName(commands.cog):
    ....

    async def get_all_parts(self, interaction, current):
        await interaction.response.defer()
        choices = []
        finalchoices = []
        partType = interaction.namespace.part_type
        for part in partTypeData["options"]:
            choices.append(part)

        for choice in choices:
            if current.lower() in choice.lower():
                finalchoices.append(app_commands.Choice(name=choice, value=choice))
        return finalchoices
        
    @app_commands.command()
    @app_commands.autocomplete(part_type=get_part_types, part_name=get_all_parts)
    async def command(self, interaction, part_type:str, part_name:str):
 ```
